pkg/scheduler/apis/config/v1alpha1/defaults.go

```diff
@@ -41,11 +41,6 @@ func SetDefaults_KubeSchedulerConfiguration(obj *kubeschedulerconfigv1alpha1.Kub
 		obj.SchedulerName = &val
 	}
 
-	if obj.HardPodAffinitySymmetricWeight == nil {
```

Comment:
The alternative would be to always set the PluginConfig here. I just liked better having a clean PluginConfig when you want defaults.

I don't think it's far from straightforward.

Reply:
by removing this default, haven't you effectively changed the behavior of existing v1alpha1 config files? previously, it would get a default weight of 1. now, it remains nil and the conversion-time setting to append to plugin args isn't exercised, right?

Reply:
Correct.
An alternative would be to keep these lines and only skip from conversion if it's the default.

What I was going for in v1alpha2 is to leave nil for default plugin arguments.

Reply:
Or we could just always fill the PluginConfig for v1alpha1, but not for v1alpha2. WDYT?

Reply:
*has* the default behavior changed for v1alpha1 config files that do not specify a HardPodAffinitySymmetricWeight field?

Reply:
No, it hasn't changed.

Reply:
Ok, that wasn't clear. As long as the resulting scheduler functions the same, that's reasonable. Moving it out of API defaulting makes that hard to test (and will make it easy to accidentally change defaults in the future without realizing, which isn't great)

Reply:
The flexibility of the new plugins API makes it a bit harder for validation and defaults. But #87784 and #88093 should alleviate some of that.